src/core/include/openvino/core/descriptor/tensor.hpp

```diff
@@ -40,6 +40,12 @@ OPENVINO_DEPRECATED("set_ov_tensor_legacy_name() is deprecated. Please don't use
 OPENVINO_API
 void set_ov_tensor_legacy_name(Tensor& tensor, const std::string& tensor_name);
 
+// To change Tensor element type please change the Parameter type.
+OPENVINO_API void set_element_type(Tensor& tensor, const element::Type& elemenet_type);
+
+// To change Tensor type please change the Parameter type.
+OPENVINO_API void set_tensor_type(Tensor& tensor, const element::Type& element_type, const PartialShape& pshape);
```

Comment:
Is `OPENVINO_API` required? It makes functions public for users, but original intention was to hide it

Reply:
@vurusovs  Without "OPENVINO_API" will cause some linker error, as following:
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/4ae0e4d0-1706-4525-a0ae-48b2c38a5f5b)
 
I also tried to move the implement of "remember_input_data_types()" & "restore_input_data_types()" from hpp to cpp, but get the same error. https://github.com/zhaixuejun1993/openvino/blob/1cf81e8400680eb49ce9245ea4ccf85216987b60/src/common/transformations/include/ov_ops/type_relaxed.hpp#L83-L125

I don't know why for "class ov::op::TypeRelaxedBase" & "class ov::op::TemporaryReplaceOutputType" we need to put the implementation in hpp.